velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -382,8 +396,14 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
           VELOX_CHECK_GE(children.size(), 1);
           if (children.size() == 1 && name != "array" &&
               name != schema[parentSchemaIdx].name + "_tuple") {
-            auto type =
-                TypeFactory<TypeKind::ARRAY>::create(children[0]->type());
+            auto elementType = children[0]->type();
+            if (requestedType) {
+              auto arrayType = requestedType->asArray();
+              if (arrayType.elementType()->isVarchar()) {
+                elementType = VARCHAR();
```

Comment:
Why do we need to do this?

Reply:
As explained above,This is used to handle whether array elements are treated as strings.

Reply:
What's the value of `children[0]->type()` in this case?

Reply:
`children[0]->type()` is parsed from `fileMetaData_->schema`. For example, the schema of a Parquet file is as follows:
```
message schema {
  optional group a (LIST) {
    repeated group list {
      optional binary element (UTF8);
    }
  }
}
```
In this scenario, `children[0]->type()` is `VARCHAR`.  If the parquet schema is as follows:
```
message schema {
  optional group a (LIST) {
    repeated group list {
      optional binary element;
    }
  }
}
```
and `options_.fileSchema()` is `ROW(a ARRAY[VARCHAR])`, then  `children[0]->type()` is `VARBINARY`,  The following code will set `elementType` to `VARCHAR`. This implements the read binary as string semantics.
```
auto elementType = children[0]->type();
if (requestedType) {
    VELOX_CHECK(requestedType->isArray())
    auto arrayType = requestedType->asArray();
    if (arrayType.elementType()->isVarchar()) {
        elementType = VARCHAR();
    }
}
```